starport/services/network/blockchain.go

```diff
@@ -151,58 +159,89 @@ func (b *Blockchain) Publish(ctx context.Context, options ...CreateOption) error
 
 		genesis, genesisHash, err = genesisAndHashFromURL(ctx, o.genesisURL)
 		if err != nil {
-			return err
+			return 0, 0, err
 		}
 
 		if !o.noCheck {
 			if !b.isInitialized {
 				if err := b.Init(ctx); err != nil {
-					return err
+					return 0, 0, err
 				}
 			}
 
 			genesisPath, err := b.chain.GenesisPath()
 			if err != nil {
-				return err
+				return 0, 0, err
 			}
 
 			if err := os.WriteFile(genesisPath, genesis, 0666); err != nil {
-				return err
+				return 0, 0, err
 			}
 
 			commands, err := b.chain.Commands(ctx)
 			if err != nil {
-				return err
+				return 0, 0, err
 			}
 
 			if err := commands.ValidateGenesis(ctx); err != nil {
-				return err
+				return 0, 0, err
 			}
 		}
 	}
 
 	chainID, err := b.chain.ID()
 	if err != nil {
-		return err
+		return 0, 0, err
 	}
 
+	coordinatorAddress := b.builder.account.Address(SPNAddressPrefix)
+	campaignID = o.campaignID
+
 	_, err = profiletypes.
 		NewQueryClient(b.builder.cosmos.Context).
 		CoordinatorByAddress(ctx, &profiletypes.QueryGetCoordinatorByAddressRequest{
-			Address: b.builder.account.Address(SPNAddressPrefix),
+			Address: coordinatorAddress,
 		})
 
 		// TODO check for not found and only then create a new coordinator, otherwise return the err.
 	if err != nil {
 		msgCreateCoordinator := profiletypes.NewMsgCreateCoordinator(
-			b.builder.account.Address(SPNAddressPrefix),
+			coordinatorAddress,
 			"",
 			"",
 			"",
 		)
 		if _, err := b.builder.cosmos.BroadcastTx(b.builder.account.Name, msgCreateCoordinator); err != nil {
-			return err
+			return 0, 0, err
+		}
+	}
+
+	if campaignID != 0 {
+		_, err = campaigntypes.
+			NewQueryClient(b.builder.cosmos.Context).
+			Campaign(ctx, &campaigntypes.QueryGetCampaignRequest{
+				Id: o.campaignID,
+			})
+		if err != nil {
+			return 0, 0, err
 		}
+	} else {
+		msgCreateCampaign := campaigntypes.NewMsgCreateCampaign(
+			coordinatorAddress,
+			"default",
```

Comment:
We should use the chain name

Reply:
you mean the chain id or something else, 78c35df

Reply:
Chain id is better but still include the network number `<name>-<n>` which should be unrelated to the campaign
IMO the best would be to get the chain name https://github.com/tendermint/starport/blob/0192431428f7d97a1a4015bf945331a2d983ecf0/starport/services/chain/app.go#L31